src/workerd/jsg/async-context.h

```diff
@@ -83,7 +83,7 @@ class AsyncContextFrame final: public Wrappable {
     bool isDead() const { return dead; }
     inline uint hashCode() const { return hash; }
     inline bool operator==(const StorageKey& other) const {
-      return hash == other.hash;
+      return this == &other;
```

Comment:
Can you also fix line 184 below, which compares hash codes but should be checking equality?

Should we grep the codebase for `hash.*==.*hash` to find other instances of this? I seem to recall seeing others.

Reply:
ah right, good catch. May be worthwhile to grep for others but let's do that in a separate PR to keep this one focused on StorageKey.